classes/file_storage.php

```diff
@@ -546,19 +586,46 @@ private static function exportFileTree($target, $contextid, $filearea, $filepath
         $fs = get_file_storage();
         $files = $fs->get_directory_files($contextid, 'mod_hvp', $filearea, $itemid, $filepath, true);
 
-        foreach ($files as $file) {
-            // Correct target path for file.
-            $path = $target . str_replace($filepath, '/', $file->get_filepath());
-
-            if ($file->is_directory()) {
-                // Create directory.
-                $path = rtrim($path, '/');
-                if (!file_exists($path)) {
-                    mkdir($path, 0777, true);
+        if ($filearea === 'libraries' && $fs->file_exists(
+            $contextid,
+            'mod_hvp',
+            $filearea,
+            $itemid,
+            $filepath,
+            'lib-export.zip'
+        )) {
+            // Libraries may have a precompiled zip to extract(
+            $file = $fs->get_file(
+                $contextid,
+                'mod_hvp',
+                $filearea,
+                $itemid,
+                $filepath,
+                'lib-export.zip'
```

Comment:
just wondering whether we should stash these in a completely different filearea eg 'library-archives' so it can never leak or clash

Reply:
also can you combine the file_exists and get_file into one?